### ISSUE TITLE
Nett.AspNet's AddTomlFile() throws System.InvalidOperationException: 'Unexpected'

### ISSUE THREAD

**Issue by jotosmurf** [paiden/Nett#38]
Giving the following ASP.NET Core 2 code (pasting the relevant bits):

```csharp
using Nett;
using Nett.AspNet;

public class Program
{
    public static void Main(string[] args) {
        BuildWebHost(args).Run();
    }

    public static IWebHost BuildWebHost(string[] args) => WebHost.CreateDefaultBuilder(args)
            .ConfigureAppConfiguration((builderContext, config) => {

                var path = Path.Combine(builderContext.HostingEnvironment.ContentRootPath, "config.toml");
                config.AddTomlFile(path, optional: false, reloadOnChange: true);

            })
            .UseStartup<Startup>()
            .Build();
}
```

Results in the following stack trace:

```csharp
System.InvalidOperationException occurred
  HResult=0x80131509
  Message=Unexpected
  Source=<Cannot evaluate the exception source>
  StackTrace:
   at Microsoft.Extensions.Configuration.FileConfigurationProvider.Load(Boolean reload)
   at Microsoft.Extensions.Configuration.FileConfigurationProvider.Load()
   at Microsoft.Extensions.Configuration.ConfigurationRoot..ctor(IList`1 providers)
   at Microsoft.Extensions.Configuration.ConfigurationBuilder.Build()
   at Microsoft.AspNetCore.Hosting.WebHostBuilder.BuildCommonServices(AggregateException& hostingStartupErrors)
   at Microsoft.AspNetCore.Hosting.WebHostBuilder.Build()
   at Program.BuildWebHost(String[] args) in Program.cs:line 22
   at Program.Main(String[] args) in Program.cs:line 19
```

Please note that manually reading the file actually works.

**Comment by paiden** [comment]
I just tried your code fragment and it runs without any issues for me.

I do not see any Nett related method in the call stack.

What is the exception message?

**Comment by jotosmurf** [comment]
@paiden Unfortunately there are no inner exceptions nor aggregate exceptions. 

I used the [ConfigurationProvider by Bojan Rajkovic](https://github.com/bojanrajkovic/TomlConfigurationProvider) which worked for ASP.NET Core 2.0. 

However upgrading to ASP.NET Core 2.2 Bojan's provider stopped working. I tried your implementation again on ASP.NET Core 2.2 and it still throws the same Unexpected exception without any clues as to where it fails exactly.
